ncm-systemd/src/main/perl/Systemd/Service/Unit.pm

```diff
@@ -462,6 +459,14 @@ sub configured_units
         $units{$detail->{name}} = $detail;
     }
 
+    if (@tryrestart) {
+        # It is ok to do the conditional restart here. This does not change state.
```

Comment:
This part of the code runs before the enabling/disabling/start/stop code of the units.
Conditional restarts do not changes the states: previously running (should) still be running; stopped is still stopped. 
But indeed, modiifcations to the unitfile can cause a unit not to get started. But that can be said about anything that you can configure.